README.md

```diff
@@ -247,10 +252,88 @@ If you are running in an "air-gapped" environment, you may need to download the
+1. The password to your Postgres database. By default, this is located at
+   `/hab/svc/builder-datastore/config/pwfile`
+1. A fresh backup of the two databases present in the On-Premise Depot,
+   `builder_sessionsrv` and `builder_originsrv`. You can create such a backup
+   with `pg_dump`:
+
+   ```shell
+   PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile) hab pkg exec core/postgresql pg_dump -h 127.0.0.1 -p 5432 -U hab builder_originsrv > builder-originsrv.sql
+   ```
+
+### Migration
+1. Stop the `builder-api` service (e.g. run `systemctl stop hab-sup` on the
+   `builder-api` node) so that incoming HTTP requests don't write
+   data to the database during the migration.
+1. Upgrade the `builder-sessionsrv` and `builder-originsrv` services to the
+   newest version available in the `on-prem-stable` channel.
+   `habitat/builder-originsrv/7568/20180816162314` and
+   `habitat/builder-sessionsrv/7568/20180816162313` are the earliest packages
+   that contain the new database code. One way to install these would be via
+   `hab`:
+
+   ```shell
+   hab pkg install -c on-prem-stable habitat/builder-originsrv
+   ```
+1. Once `builder-sessionsrv` and `builder-originsrv` have been upgraded to
+   the new version, you will likely see lines like this in your logs:
+   `Shard migration hasn't been completed successfully` repeated over and over
+   again, as the supervisor tries to start the new service, but the service
+   dies because the migration hasn't been run.
+1. If you want to be extra sure that you're in a good spot to perform the
+   migration, log into the Postgres console and verify that you have empty
+   tables in the `public` schema. A command to do this might look like:
+
+   ```shell
+   PGPASSWORD=$(cat /hab/svc/builder-datastore/config/pwfile) hab pkg exec
+   core/postgresql psql -h 127.0.0.1 -p 5432 -U hab builder_originsrv`.
+   ```
+
+   That should drop you into a prompt where you can type `\d` and hopefully see
+   a list of tables where the schema says `public`. If you try to select data
+   from any of those tables, they should be empty.
+1. Now you're ready to migrate the data itself. The following command will do
+   that for `builder-originsrv`:
+
+   ```shell
+   PGPASSWORD=$(cat
+   /hab/svc/builder-datastore/config/pwfile) /path/to/migrate.sh originsrv
+   migrate
+   ```
+
+   After confirming that you have fresh database backups, the script
+   should run and at the end, you should see several notices that everything is
+   great, row counts check out, and your database has been marked as migrated.
+1. Do the same migration for `builder-sessionsrv`.
+
+   ```shell
+   PGPASSWORD=$(cat
+   /hab/svc/builder-datastore/config/pwfile) /path/to/migrate.sh sessionsrv
+   migrate
+   ```
+
+1. Restart both `builder-originsrv` and `builder-sessionsrv` and the logs
+   should look normal again.
+1. Start the `builder-api` service back up.
```

Comment:
This step can probably go away - after the re-install, no data will get written until migration completes since no service will be up to write the data with.